### DISCUSSION TITLE
Cannot parse regex negative lookbehind

### DISCUSSION THREAD

**Discussion by prettydiff** [eslint/eslint#13810] [Q&A] [answered]
**Tell us about your environment**

* **ESLint Version:** 7.12.1
* **Node Version:** 14.4.0
* **npm Version:** 6.14.5

**What parser (default, `@babel/eslint-parser`, `@typescript-eslint/parser`, etc.) are you using?**
Default parser.  Maybe typescrip-eslint/parser.  I do not have babel installed.

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
{
    "env": {
        "browser":true,
        "es6":true,
        "node":true
    },
    "extends":"eslint:recommended",
    "parserOptions": {
        "sourceType":"module"
    },
    "rules": {
        "curly":"error",
        "eqeqeq":"error",
        "max-params":["error",3],
        "one-var":["error","always"],
        "operator-assignment":["error","never"],
        "no-console":"error",
        "no-continue":"error",
        "no-control-regex":0,
        "no-else-return":"error",
        "no-empty":"error",
        "no-extra-semi":"error",
        "no-invalid-this":"error",
        "no-new":"error",
        "no-plusplus":"error",
        "no-restricted-syntax":["error",{
            "selector":"ThisExpression",
            "message":"Use explicit reference names instead of 'this'."
        }, {
            "selector":"ThisStatement",
            "message":"Use explicit reference names instead of 'this'."
        }, {
            "selector":"ClassDeclaration",
            "message":"No classes. Inheritance imposes unseen complexity upon code extension."
        }, {
            "selector":"ClassExpression",
            "message":"No classes. Inheritance imposes unseen complexity upon code extension."
        }, {
            "selector":"TryStatement",
            "message":"Do not use try/catch blocks as these are performance bottlenecks."
        }],
        "no-var":"error",
        "semi":["error","always"],
        "strict":["error","function"],
        "wrap-iife":["error","outside"],
        "wrap-regex":"error"
    }
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
myString.replace(/(?<!\\)(\\u[0-9a-f]{4})+/g, someFunction)
```

```bash
eslint ./
```

**What did you expect to happen?**
No errors

**What actually happened? Please include the actual, raw output from ESLint.**
Parse error.  ESLint cannot parse the first regex group `(?<!\\)`

Please see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions/Assertions

**Are you willing to submit a pull request to fix this bug?**
I am not sure.  I am not familiar with ESLint's regular expression parser.

**Comment by prettydiff** [comment]
This is frustrating in that since it's a parse error the `// eslint-disable-next-line` is ignored for that line of code.  Therefore lint always fails.

**Comment by aladdin-add** [comment, marked as answer]
To allow the usage of negative lookbehind, you need to set `ecmaVersion` >= 2018 (as it defaults to 5)

refs: https://github.com/tc39/proposal-regexp-lookbehind
